runtime/particle-execution-host.js

```diff
@@ -57,9 +52,9 @@ export class ParticleExecutionHost {
     };
 
     this._apiPort.onHandleSet = ({handle, data, particleId, barrier}) => handle.set(data, particleId, barrier);
-    this._apiPort.onHandleStore = ({handle, data, particleId}) => handle.store(data, particleId);
     this._apiPort.onHandleClear = ({handle, particleId, barrier}) => handle.clear(particleId, barrier);
-    this._apiPort.onHandleRemove = ({handle, data, particleId}) => handle.remove(data, particleId);
+    this._apiPort.onHandleStore = ({handle, data: {value, keys}, particleId}) => handle.store(value, keys, particleId);
+    this._apiPort.onHandleRemove = ({handle, data: {value, keys}, particleId}) => handle.remove(value, keys, particleId);
```

Comment:
Do you think with CRDTs that it might make sense to have hierarchical APIs at some point? i.e. we can move a lot of the handle-specific protocol into a different file that gets imported here?

Not something to do now, just an idle thought.